java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -530,9 +531,18 @@ public ArrowFlightSqlClientHandler build() throws SQLException {
                   client, new CredentialCallOption(new BearerCredentialWriter(token))));
         }
         return ArrowFlightSqlClientHandler.createNewHandler(client, options);
-
-      } catch (final IllegalArgumentException | GeneralSecurityException | IOException | FlightRuntimeException e) {
-        final SQLException originalException = new SQLException(e);
+      } catch (final IllegalArgumentException | GeneralSecurityException | IOException e) {
+        SQLException originalException = new SQLException(e);
+        if (client != null) {
+          try {
+            client.close();
+          } catch (final InterruptedException interruptedException) {
+            originalException.addSuppressed(interruptedException);
+          }
+        }
+        throw originalException;
+      } catch (final FlightRuntimeException e) {
+        SQLException originalException = RpcExceptionMapper.map(e, "Failure in connection: %s", e.toString());
```

Comment:
It's very confusing to call this originalException.

Reply:
This was created to ensure that the client had been closed after throwing the exceptions.